### PR TITLE
Amazon Bedrock - Model Throughput Provisioning

### PR DESCRIPTION
**Pull request by ferruzzi**
Adds support for provisioning model throughput on Amazon Bedrock including: operators, sensors, waiters, and triggerer, as well as unit tests and system tests for all of the above, and doc updates.

Manually tested in Breeze with the Trigger and with the Sensor:
image

Sits on top of a change in https://github.com/apache/airflow/pull/38849, please make sure that is merged before this one.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/amazon/aws/operators/bedrock.py [hunk 1]
```diff
@@ -25,7 +25,10 @@
 from airflow.exceptions import AirflowException
```
Comment (ferruzzi) [inline airflow/providers/amazon/aws/operators/bedrock.py:25]: This comment is just a safety gate to make sure this PR doesn't get merged out of order; anyone can feel free to resolve it once https://github.com/apache/airflow/pull/38849 is merged.

File: airflow/providers/amazon/aws/sensors/bedrock.py [hunk 2]
```diff
@@ -17,36 +17,89 @@
...
+    FAILURE_STATES: tuple[str, ...] = ()
+    SUCCESS_STATES: tuple[str, ...] = ()
+    FAILURE_MESSAGE = ""
+
+    aws_hook_class = BedrockHook
+    ui_color = "#66c3ff"
+
+    def __init__(
+        self,
+        deferrable: bool = conf.getboolean("operators", "default_deferrable", fallback=False),
```
Comment (vincbeck) [inline airflow/providers/amazon/aws/sensors/bedrock.py:65]: nit: missing from docstring

File: airflow/providers/amazon/aws/operators/bedrock.py (hunk 1 above)
Comment (ferruzzi) [reply airflow/providers/amazon/aws/operators/bedrock.py:25]: dependency cleared.

File: airflow/providers/amazon/aws/sensors/bedrock.py (hunk 2 above)
Comment (ferruzzi) [reply airflow/providers/amazon/aws/sensors/bedrock.py:65]: Looks like somehow I put the aws_conn_id docstring there instead.  I'll fix it.  Good catch, thanks.

File: tests/system/providers/amazon/aws/example_bedrock.py [hunk 3]
```diff
@@ -127,6 +137,7 @@ def delete_custom_model(model_name: str):
         output_data_uri=f"s3://{bucket_name}/myOutputData",
     )
     # [END howto_operator_customize_model]
+    customize_model.wait_for_completion = False
```
Comment (o-nikolas) [inline tests/system/providers/amazon/aws/example_bedrock.py]: Will this fix the stability issues we're seeing in the system test right now?

**Review by o-nikolas — APPROVED** [review]

File: tests/system/providers/amazon/aws/example_bedrock.py (hunk 3 above)
Comment (ferruzzi) [reply tests/system/providers/amazon/aws/example_bedrock.py]: I don't think so.   I am going to change the system test to use a branching operator like we discussed, but this was something that caught my eye while I was in there.

Comment (ferruzzi) [reply tests/system/providers/amazon/aws/example_bedrock.py]: That fix should be [here](https://github.com/apache/airflow/pull/38887)

File: airflow/providers/amazon/aws/operators/bedrock.py (hunk 1 above)
Comment (ferruzzi) [reply airflow/providers/amazon/aws/operators/bedrock.py:25]: I had to refactor the existing portion of the Bedrock system test, so this PR will need to be rebased on top of that and adjusted accordingly.  Do not merge until https://github.com/apache/airflow/pull/38887 is merged and rebased in.

**Review by vincbeck — APPROVED** [review]

**Comment by ferruzzi** [comment]
Dependency merged; I'll get this rebased on top of that one and merge it this afternoon.

**Comment by ferruzzi** [comment]
Gah, static check is just a merge issue, imported the sensor twice.  Easy fix.

**Pull request merged**